hail/src/main/scala/is/hail/backend/Backend.scala

```diff
@@ -1,16 +1,120 @@
 package is.hail.backend
 
+import com.sun.net.httpserver.{HttpContext, HttpExchange, HttpHandler, HttpServer}
+import java.io._
+import java.net._
+import java.nio.charset.StandardCharsets
+
+import org.json4s._
+import org.json4s.jackson.{JsonMethods, Serialization}
+
 import is.hail.asm4s._
 import is.hail.backend.spark.SparkBackend
 import is.hail.expr.ir.lowering.{TableStage, TableStageDependency}
 import is.hail.expr.ir.{CodeCacheKey, CompiledFunction, LoweringAnalyses, SortField, TableIR, TableReader}
+import is.hail.io.{BufferSpec, TypedCodecSpec}
 import is.hail.io.fs._
+import is.hail.io.plink.LoadPlink
+import is.hail.io.vcf.LoadVCF
+import is.hail.expr.ir.{IRParser, BaseIR}
 import is.hail.linalg.BlockMatrix
 import is.hail.types._
+import is.hail.types.encoded.EType
+import is.hail.types.virtual.TFloat64
+import is.hail.types.physical.PTuple
 import is.hail.utils._
 import is.hail.variant.ReferenceGenome
 
+import scala.collection.mutable
 import scala.reflect.ClassTag
+import is.hail.expr.ir.IRParserEnvironment
+
+
+case class IRTypePayload(ir: String)
+case class LoadReferencesFromDatasetPayload(path: String)
+case class FromFASTAFilePayload(name: String, fasta_file: String, index_file: String,
+    x_contigs: Array[String], y_contigs: Array[String], mt_contigs: Array[String],
+    par: Array[String])
+case class ParseVCFMetadataPayload(path: String)
+case class ImportFamPayload(path: String, quant_pheno: Boolean, delimiter: String, missing: String)
+case class ExecutePayload(ir: String, stream_codec: String, timed: Boolean)
+
+object BackendServer {
+  def apply(backend: Backend) = new BackendServer(backend)
+}
+
+class BackendServer(backend: Backend) {
```

Comment:
I feel like this file is getting rather large; does it make sense to put the HTTP stuff in a different file?

Reply:
Moved